### ISSUE TITLE
Support Roblox's pivot API

### ISSUE THREAD

**Issue by Anaminus** [Anaminus/rbxmk#67]
It would be wildly convenient if it were possible to use GetPivot and PivotTo to move models and parts around.

Considerations:
- Several properties from several classes are affected.
- Must be descriptor-agnostic. Should be similar to descriptor/AttrConfig API.
- A configuration matching Roblox is provided by default.
- Roblox can add more affected classes/properties in the future. The implementation should be simple to extend.
	- Use descriptor extensions (#65).
- Basically involves implementing Roblox's pivoting engine. Could be daunting.
	- We have a complete CFrame implementation, so it might not be so bad.
- Should have **thorough testing**!
	- Write tests for Roblox's implementation. Design them so that they can also run in rbxmk.
